restli-client/src/main/java/com/linkedin/restli/client/Request.java

```diff
@@ -299,13 +302,29 @@ public UriTemplate getUriTemplate()
    */
   protected Set<PathSpec> getFields()
   {
-    @SuppressWarnings("unchecked")
-    Set<PathSpec> fieldsSet = (Set<PathSpec>) _queryParams.get(RestConstants.FIELDS_PARAM);
-    if (fieldsSet == null)
-    {
+    Object fields = _queryParams.get(RestConstants.FIELDS_PARAM);
+    if (fields == null) {
       return Collections.emptySet();
     }
-    return fieldsSet;
+
+    if (fields instanceof Set)
```

Comment:
Maybe we can provide an API in `AbstractRequestBuilder` to have such an API for users to directly set fields parameter instead of using `setParam`. Something like

```
public void addFields(String fieldPaths)
public void addFields(PathSpec... fieldPathSpecs)
```

Ideally, we can even expose `fields(String... fieldPaths)` in generated type safe request builders to get this optimization better utilized, but that can be done later. WDYT?

Reply:
As we chatted offline, for typesafe request builders, maybe we don't want to expose application developers such advanced trick to avoid invalid pathSpec string passed in. For infra, this may be a nice feature to adoption. So the suggestion to add API in `AbstractRequestBuilder` may have value there. But this can be done in followup RB with team discussion.

Reply:
As part of Universal client, I have plans to add APIs that accept MaskTree and provide helpers so clients can build it. That would be a good trade-off between typesafety and performance.